src/EventStore.Client/EventStoreClientSettings.ConnectionString.cs

```diff
@@ -174,51 +187,73 @@ private static class ConnectionStringParser {
 						int value and >= 0 => TimeSpan.FromMilliseconds(value),
-						_ => throw new InvalidSettingException($"Invalid KeepAliveInterval: {keepAliveIntervalMs}")
+						_                  => throw new InvalidSettingException($"Invalid KeepAliveInterval: {keepAliveIntervalMs}")
 					};
 				}
 
 				if (typedOptions.TryGetValue(KeepAliveTimeout, out var keepAliveTimeoutMs)) {
 					settings.ConnectivitySettings.KeepAliveTimeout = keepAliveTimeoutMs switch {
-						-1 => Timeout_.InfiniteTimeSpan,
+						-1                 => Timeout_.InfiniteTimeSpan,
 						int value and >= 0 => TimeSpan.FromMilliseconds(value),
-						_ => throw new InvalidSettingException($"Invalid KeepAliveTimeout: {keepAliveTimeoutMs}")
+						_                  => throw new InvalidSettingException($"Invalid KeepAliveTimeout: {keepAliveTimeoutMs}")
 					};
 				}
 
-				connSettings.Insecure = !useTls;
+				settings.ConnectivitySettings.Insecure = !useTls;
 
 				if (hosts.Length == 1 && scheme != UriSchemeDiscover) {
-					connSettings.Address = hosts[0].ToUri(useTls);
-				} else {
+					settings.ConnectivitySettings.Address = hosts[0].ToUri(useTls);
+				}
+				else {
 					if (hosts.Any(x => x is DnsEndPoint))
-						connSettings.DnsGossipSeeds =
+						settings.ConnectivitySettings.DnsGossipSeeds =
 							Array.ConvertAll(hosts, x => new DnsEndPoint(x.GetHost(), x.GetPort()));
 					else
-						connSettings.IpGossipSeeds = Array.ConvertAll(hosts, x => (IPEndPoint)x);
+						settings.ConnectivitySettings.IpGossipSeeds = Array.ConvertAll(hosts, x => (IPEndPoint)x);
 				}
 
 				if (typedOptions.TryGetValue(TlsVerifyCert, out var tlsVerifyCert)) {
 					settings.ConnectivitySettings.TlsVerifyCert = (bool)tlsVerifyCert;
 				}
 
+				if (typedOptions.TryGetValue(TlsCaFile, out var tlsCaFile)) {
+					var tlsCaFilePath = Path.GetFullPath((string)tlsCaFile);
+					if (!string.IsNullOrEmpty(tlsCaFilePath) && !File.Exists(tlsCaFilePath)) {
+						throw new InvalidClientCertificateException($"Failed to load certificate. File was not found.");
+					}
+
+					try {
+						settings.ConnectivitySettings.TlsCaFile = new X509Certificate2(tlsCaFilePath);
+					} catch (CryptographicException) {
+						throw new InvalidClientCertificateException("Failed to load certificate. Invalid file format.");
+					}
+				}
+
 				settings.CreateHttpMessageHandler = CreateDefaultHandler;
 
 				return settings;
 
 				HttpMessageHandler CreateDefaultHandler() {
 #if NET
 					var handler = new SocketsHttpHandler {
-						KeepAlivePingDelay = settings.ConnectivitySettings.KeepAliveInterval,
-						KeepAlivePingTimeout = settings.ConnectivitySettings.KeepAliveTimeout,
+						KeepAlivePingDelay             = settings.ConnectivitySettings.KeepAliveInterval,
+						KeepAlivePingTimeout           = settings.ConnectivitySettings.KeepAliveTimeout,
 						EnableMultipleHttp2Connections = true,
 					};
 
+					var sslOptions = new SslClientAuthenticationOptions();
```

Comment:
I think we can still simplify the `CreateDefaultHandler` method here and in `ChannelFactory` further, particularly around the duplicate check `if (settings.ConnectivitySettings is { TlsCaFile: not null, Insecure: false }) {`

What would you think of something like this?
```csharp
				HttpMessageHandler CreateDefaultHandler() {
					var configureClientCert = settings.ConnectivitySettings is { TlsCaFile: not null, Insecure: false };
#if NET
					var handler = new SocketsHttpHandler {
						KeepAlivePingDelay             = settings.ConnectivitySettings.KeepAliveInterval,
						KeepAlivePingTimeout           = settings.ConnectivitySettings.KeepAliveTimeout,
						EnableMultipleHttp2Connections = true,
					};

					if (configureClientCert)
						handler.SslOptions.ClientCertificates = [settings.ConnectivitySettings.TlsCaFile!];

					if (!settings.ConnectivitySettings.TlsVerifyCert) {
						handler.SslOptions.RemoteCertificateValidationCallback = delegate { return true; };
					}
#else
					var handler = new WinHttpHandler {
						TcpKeepAliveEnabled = true,
						TcpKeepAliveTime = settings.ConnectivitySettings.KeepAliveTimeout,
						TcpKeepAliveInterval = settings.ConnectivitySettings.KeepAliveInterval,
						EnableMultipleHttp2Connections = true
					};

					if (configureClientCert)
						handler.ClientCertificates.Add(settings.ConnectivitySettings.TlsCaFile!);

					if (!settings.ConnectivitySettings.TlsVerifyCert) {
						handler.ServerCertificateValidationCallback = delegate { return true; };
					}
#endif
					return handler;
				}
			}
```

Reply:
Yes, that's better.